src/BlockTypeRegistry.h

```diff
@@ -0,0 +1,158 @@
+#pragma once
+
+
+
+
+
+#include <map>
+
+
+
+
+
+// fwd:
+class BlockState;
+
+
+
+
+
+/** Complete information about a single block type.
+The BlockTypeRegistry uses this structure to store the registered information. */
+class BlockInfo
+{
+public:
+
+	/** Callback is used to query block hints dynamically, based on the current BlockState.
+	Useful for example for redstone lamps that can be turned on or off. */
+	using HintCallback = std::function<AString(const AString & aTypeName, const BlockState & aBlockState)>;
+
+
+	/** Creates a new instance with the specified BlockTypeName and handler / hints / callbacks.
+	aPluginName specifies the name of the plugin to associate with the block type (to allow unload / reload). */
+	BlockInfo(
+		const AString & aPluginName,
+		const AString & aBlockTypeName,
+		std::shared_ptr<cBlockHandler> aHandler,
+		const std::map<AString, AString> & aHints = std::map<AString, AString>(),
+		const std::map<AString, HintCallback> & aHintCallbacks = std::map<AString, HintCallback>()
+	);
+
+
+	/** Retrieves the value associated with the specified hint for this specific BlockTypeName and BlockState.
+	Queries callbacks first, then hints if a callback doesn't exist.
+	Returns an empty string if hint not found at all. */
+	AString hintValue(
```

Comment:
It could be useful to have `hintValueI`, `hintValueD`, `hintValueF` methods like the `cIniFile` class.

Reply:
Could be useful, although there are many more usages (`LightValue` hint will most likely be converted to a `UInt8`, ...)

Reply:
I won't add it now, but it could be added later on.